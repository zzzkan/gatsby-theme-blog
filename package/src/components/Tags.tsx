import React from "react"
import { Link as GatsbyLink } from "gatsby"
import { Link, Tag, Wrap, WrapItem } from "@chakra-ui/react"

export type TagsProps = {
  tags: readonly string[] | null
}

const Tags: React.FC<TagsProps> = ({ tags }) => {
  if (tags == null || tags.length === 0) return null
  return (
    <Wrap>
      {tags.map((tag) => (
        <WrapItem key={tag}>
          <Link
            as={GatsbyLink}
            to={`/tags/${tag}`}
            _hover={{ textDecoration: "none" }}
          >
            <Tag
              size={"sm"}
              variant={"solid"}
              color={"onTintText"}
              background={"tint"}
            >
              {tag}
            </Tag>
          </Link>
        </WrapItem>
      ))}
    </Wrap>
  )
}

export default Tags
