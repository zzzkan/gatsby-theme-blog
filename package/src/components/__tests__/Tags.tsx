import React from "react"
import { render } from "@testing-library/react"
import { Tags } from "../Tags"

describe("Tags component", () => {
  test("snapshot", () => {
    const { asFragment } = render(
      <Tags
        tags={[
          { slug: "/tags/tag1", name: "tag1" },
          { slug: "/tags/tag2", name: "tag2" },
          { slug: "/tags/tag3", name: "tag3" },
          { slug: "/tags/tag4", name: "tag4" },
          { slug: "/tags/tag5", name: "tag5" },
        ]}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("should not render anything if props is null", () => {
    const { container } = render(<Tags tags={null} />)
    expect(container.firstChild).toBeNull()
  })

  test("should not render anything if props is []", () => {
    const { container } = render(<Tags tags={[]} />)
    expect(container.firstChild).toBeNull()
  })

  test("should render tag as /tags/* link", () => {
    const { getByRole } = render(
      <Tags tags={[{ slug: "/tags/tag", name: "tag" }]} />
    )
    expect(getByRole("link")).toHaveAttribute("href", "/tags/tag")
  })

  test("should render tag as /base/tags/* link", () => {
    const { getByRole } = render(
      <Tags tags={[{ slug: "/base/tags/tag", name: "tag" }]} />
    )
    expect(getByRole("link")).toHaveAttribute("href", "/base/tags/tag")
  })

  test("should render multiple tag", () => {
    const { getAllByRole } = render(
      <Tags
        tags={[
          { slug: "/tags/tag1", name: "tag1" },
          { slug: "/tags/tag2", name: "tag2" },
          { slug: "/tags/tag3", name: "tag3" },
          { slug: "/tags/tag4", name: "tag4" },
          { slug: "/tags/tag5", name: "tag5" },
          { slug: "/tags/tag6", name: "tag6" },
          { slug: "/tags/tag7", name: "tag7" },
          { slug: "/tags/tag8", name: "tag8" },
          { slug: "/tags/tag9", name: "tag9" },
          { slug: "/tags/tag10", name: "tag10" },
        ]}
      />
    )
    expect(getAllByRole("link").length).toBe(10)
  })
})
