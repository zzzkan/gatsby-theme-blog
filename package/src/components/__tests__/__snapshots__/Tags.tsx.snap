// Vitest Snapshot v1

exports[`Tags component > snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  overflow: hidden;
}

.emotion-1 {
  --chakra-wrap-x-spacing: 0.5rem;
  --chakra-wrap-y-spacing: 0.5rem;
  --wrap-x-spacing: calc(var(--chakra-wrap-x-spacing) / 2);
  --wrap-y-spacing: calc(var(--chakra-wrap-y-spacing) / 2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style-type: none;
  padding: 0px;
  margin: calc(var(--wrap-y-spacing) * -1) calc(var(--wrap-x-spacing) * -1);
}

.emotion-1>*:not(style) {
  margin: var(--wrap-y-spacing) var(--wrap-x-spacing);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-3:hover,
.emotion-3[data-hover] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: top;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  color: onTintText;
  background: tint;
}

<div
    class="chakra-wrap emotion-0"
  >
    <ul
      class="chakra-wrap__list emotion-1"
    >
      <li
        class="chakra-wrap__listitem emotion-2"
      >
        <a
          class="chakra-link emotion-3"
          href="/tags/tag1"
        >
          <span
            class="emotion-4"
          >
            tag1
          </span>
        </a>
      </li>
      <li
        class="chakra-wrap__listitem emotion-2"
      >
        <a
          class="chakra-link emotion-3"
          href="/tags/tag2"
        >
          <span
            class="emotion-4"
          >
            tag2
          </span>
        </a>
      </li>
      <li
        class="chakra-wrap__listitem emotion-2"
      >
        <a
          class="chakra-link emotion-3"
          href="/tags/tag3"
        >
          <span
            class="emotion-4"
          >
            tag3
          </span>
        </a>
      </li>
      <li
        class="chakra-wrap__listitem emotion-2"
      >
        <a
          class="chakra-link emotion-3"
          href="/tags/tag4"
        >
          <span
            class="emotion-4"
          >
            tag4
          </span>
        </a>
      </li>
      <li
        class="chakra-wrap__listitem emotion-2"
      >
        <a
          class="chakra-link emotion-3"
          href="/tags/tag5"
        >
          <span
            class="emotion-4"
          >
            tag5
          </span>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
